src/api/c/wrap.cpp

```diff
@@ -19,92 +19,91 @@ using af::dim4;
-                break;
-            case u64:
-                output =
-                    wrap<uintl>(in, ox, oy, wx, wy, sx, sy, px, py, is_column);
-                break;
-            case s16:
-                output =
-                    wrap<short>(in, ox, oy, wx, wy, sx, sy, px, py, is_column);
-                break;
-            case u16:
-                output =
-                    wrap<ushort>(in, ox, oy, wx, wy, sx, sy, px, py, is_column);
-                break;
-            case u8:
-                output =
-                    wrap<uchar>(in, ox, oy, wx, wy, sx, sy, px, py, is_column);
-                break;
-            case b8:
-                output =
-                    wrap<char>(in, ox, oy, wx, wy, sx, sy, px, py, is_column);
-                break;
-            default: TYPE_ERROR(1, type);
+        // clang-format off
+        switch(in_type) {
+            case f32: wrap<float  >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case f64: wrap<double >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case c32: wrap<cfloat >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case c64: wrap<cdouble>(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case s32: wrap<int    >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case u32: wrap<uint   >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case s64: wrap<intl   >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case u64: wrap<uintl  >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case s16: wrap<short  >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case u16: wrap<ushort >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case u8:  wrap<uchar  >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            case b8:  wrap<char   >(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column);  break;
+            default:  TYPE_ERROR(1, in_type);
         }
-        std::swap(*out, output);
+        // clang-format on
     }
     CATCHALL;
 
     return AF_SUCCESS;
 }
+
+af_err af_wrap(af_array* out, const af_array in,
+               const dim_t ox, const dim_t oy,
+               const dim_t wx, const dim_t wy,
+               const dim_t sx, const dim_t sy,
+               const dim_t px, const dim_t py,
+               const bool is_column) {
+    return af_wrap_common(out, in, ox, oy, wx, wy, sx, sy, px, py, is_column,
+                          true);
+}
+
+af_err af_wrap_v2(af_array* out, const af_array in,
+                  const dim_t ox, const dim_t oy,
+                  const dim_t wx, const dim_t wy,
+                  const dim_t sx, const dim_t sy,
+                  const dim_t px, const dim_t py,
+                  const bool is_column) {
+    if (out == 0) return AF_ERR_ARG;
```

Comment:
I think you should do the same conversion you did in the approx_v2 functions here for the error messages.